File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.java
import java.util.Arrays;
import java.util.Stack;

public class Solution {
    public int[] solution(int[] arr) {
        Stack<Integer> stk = new Stack<>();
        int i = 0;

        while (i < arr.length) {
            int current = arr[i++];
            if (!stk.isEmpty() && stk.peek().equals(current)) {
                stk.pop();
            } else {
                stk.push(current);
            }
        }

        int size = stk.size();
        return size > 0 ? createResultArray(stk, size) : new int[]{-1};
    }

    private int[] createResultArray(Stack<Integer> stk, int size) {
        int[] answer = new int[size];
        for (int j = size - 1; j >= 0; j--) {
            answer[j] = stk.pop();
        }
        return answer;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        int[] arr = {0, 1, 1, 1, 0};
        int[] answer = sol.solution(arr);
        System.out.println("Arrays.toString(answer) = " + Arrays.toString(answer));
    }
}

File: JavaScript/프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.js
function solution (arr) {
    const answer = [];
    let i = 0;

    while (i < arr.length) {
        if (arr.length === 0) {
            answer.push(arr[i]);
            i++;
        } else if (answer[answer.length - 1] === arr[i]){
            answer.pop();
            i++;
        } else {
            answer.push(arr[i]);
            i++;
        }
    }
    
    if (answer.length === 0){
        answer.push(-1);
    }

    return answer;
};